Change V3.1 (#358)
admin-shell-io/aas-specs: documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc

== Predefined Simple Data Types

The metamodel of the Asset Administration Shell uses some of the predefined simple data types of the  link:https://www.w3.org/XML/Core/[XML Schema Definition] (XSD) 
as its basic data types footnote:[https://www.w3.org/XML/Core/, former https://www.w3.org/XML/Schema]. See Table 7 for an overview of the used types. Their definition is outside the scope of this document.

The meaning and format of xsd types is specified in https://www.w3.org/XML/Schema. The simple type "langString" is specified in the Resource Description Framework (RDF) footnote:[see: https://www.w3.org/TR/rdf11-concepts/].

See Clause 5.3.12.6 for constraints on types.

.Simple Data Types Used in Metamodel
[cols="10%,19%,38%,33%",options="header",]
|===
|Source |Basic Data Type |Value Range |Sample Values

|xsd |string |Character string (but not all Unicode character strings) |"Hello world", "Καλημέρα κόσμε", +
"ハローワールド""

|xsd |base64Binary |base64-encoded binary data |"a3Vtb3dhc2hlcmU="

|xsd |boolean |true, false |true, false

> |xsd |anyUri | Represents an Internationalized Resource Identifier Reference (IRI) conformant to https://www.ietf.org/rfc/rfc3987.txt[RFC 3987], can be absolute or relative, and may have an optional fragment identifier  a| {blank}./Specification.pdf

Reviewer: This is inconsistent to line 203 - where we write:

...as URIs per [RFC 3986] and its successor(s).

Please replace with:
|xsd |anyUri | Represents an Uniform Resource Identifier (URI) conformant to https://www.ietf.org/rfc/rfc3986[RFC 3986] and its successor(s), can be absolute or relative, and may have an optional fragment identifier  a| {blank}./Specification.pdf

Author: You are right. Resolved with https://github.com/admin-shell-io/aas-specs/pull/358/commits/d59781308ee906a9e37533adeef4b1408ff65c9f

== Predefined Simple Data Types

The metamodel of the Asset Administration Shell uses some of the predefined simple data types of the  link:https://www.w3.org/XML/Core/[XML Schema Definition] (XSD) 

> as its basic data types footnote:[https://www.w3.org/XML/Core/, former https://www.w3.org/XML/Schema]. See Table 7 for an overview of the used types. Their definition is outside the scope of this document.

Reviewer 2: ```suggestion
as its basic data types footnote:[https://www.w3.org/XML/Core/, former https://www.w3.org/TR/xmlschema-2/#built-in-datatypes]. See Table 7 for an overview of the used types. Their definition is outside the scope of this document.
```

Reviewer 2: https://www.w3.org/XML/Schema points directly to XML Schema 1.1

The meaning and format of xsd types is specified in https://www.w3.org/XML/Schema. The simple type "langString" is specified in the Resource Description Framework (RDF) footnote:[see: https://www.w3.org/TR/rdf11-concepts/].

See Clause 5.3.12.6 for constraints on types.

.Simple Data Types Used in Metamodel
[cols="10%,19%,38%,33%",options="header",]
|===
|Source |Basic Data Type |Value Range |Sample Values

> |xsd |https://www.w3.org/TR/xmlschema2/#string[string] |Character string (but not all Unicode character strings) |"Hello world", "Καλημέρα κόσμε", +

Reviewer 2: ```suggestion
|xsd |https://www.w3.org/TR/xmlschema-2/#string[string] |Character string (but not all Unicode character strings) |"Hello world", "Καλημέρα κόσμε", +
```

"ハローワールド""

> |xsd |https://www.w3.org/TR/xmlschema2/#base64Binary[base64Binary] |base64-encoded binary data |"a3Vtb3dhc2hlcmU="

Reviewer 2: ```suggestion
|xsd |https://www.w3.org/TR/xmlschema-2/#base64Binary[base64Binary] |base64-encoded binary data |"a3Vtb3dhc2hlcmU="
```

|xsd |https://www.w3.org/TR/xmlschema2/#boolean[boolean] |true, false |true, false

|xsd |https://www.w3.org/TR/xmlschema2/#anyURI[anyUri] | "[anyURI supports a] wide range of internationalized resource identifiers can be specified when an anyURI is called for, and still be understood as URIs per https://www.w3.org/TR/xmlschema11-2/#RFC3986[RFC 3986] and its successor(s)." 

It can be absolute or relative, and may have an optional fragment identifier  a| {blank}./Specification.pdf

file:c:/local/Specification.pdf

\http://www.example.org

FTP://unicode.org

> |xsd |https://www.w3.org/TR/xmlschema2/#dateType[dateType] |Date and time with or without time zone |"2000-01-01T14:23:00", +

Reviewer 2: ```suggestion
|xsd |https://www.w3.org/TR/xmlschema-2/#d0e11239[dateType] |Date and time with or without time zone |"2000-01-01T14:23:00", +
```

Reviewer 2: ```suggestion
|xsd |https://www.w3.org/TR/xmlschema-2/#dateTime[dateTime] |Date and time with or without time zone |"2000-01-01T14:23:00", +
```

> |xsd |https://www.w3.org/TR/xmlschema2/#dateType[dateType] |Date and time with or without time zone |"2000-01-01T14:23:00", +

Reviewer 2: ```suggestion
|xsd |https://www.w3.org/TR/xmlschema-2/#d0e11239[dateType] |Date and time with or without time zone |"2000-01-01T14:23:00", +
```

"2000-01-01T14:23:00.66372+14:00"footnote:[Corresponds to xs:dateTimeStamp in XML Schema 1.1]

> |xsd |https://www.w3.org/TR/xmlschema2/#duration[duration] |Duration of time |"-P1Y2M3DT1H", +

Reviewer 2: ```suggestion
|xsd |https://www.w3.org/TR/xmlschema-2/#duration[duration] |Duration of time |"-P1Y2M3DT1H", +
```

.2+e|[[NameType]]NameType 2+| `\https://admin-shell.io/aas/3/0/NameType`
a|_string_ with max 128 and min 1 characters |"ManufacturerPartId"

.2+e|[[PathType]]PathType 2+| `\https://admin-shell.io/aas/3/1/PathType`
a|

_anyURI_

with max 2024 and min 1 characters

====

> "[anyURI supports a] wide range of internationalized resource identifiers can be specified when an anyURI is called for, and still be understood as URIs per https://www.w3.org/TR/xmlschema11-2/#RFC3986[RFC 3986] and its successor(s)." 

Reviewer 2: ```suggestion
"[anyURI supports a] wide range of internationalized resource identifiers can be specified when an anyURI is called for, and still be understood as URIs per https://www.w3.org/TR/xmlschema-2/#RFC2396[RFC 2396] and https://www.w3.org/TR/xmlschema-2/#RFC2732[RFC 2732] ." 
```

> Source: https://www.w3.org/TR/xmlschema11-2/#anyURI[W3C XML Schema Definition Language (XSD) 1.1 Part 2: Datatypes]

Reviewer 2: ```suggestion
Source: https://www.w3.org/TR/xmlschema-2/#anyURI[W3C XML Schema Definition Language (XSD) 1.0 Part 2: Datatypes]
```
